app/src/main/java/com/krachkovsky/it_anews/ui/fragments/NewsListFragment.kt

```diff
@@ -0,0 +1,220 @@
+package com.krachkovsky.it_anews.ui.fragments
+
+import android.graphics.Rect
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AbsListView
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.snackbar.Snackbar
+import com.krachkovsky.it_anews.adapter.NewsAdapter
+import com.krachkovsky.it_anews.databinding.FragmentListBinding
+import com.krachkovsky.it_anews.models.NewsEverything
+import com.krachkovsky.it_anews.models.PagingData
+import com.krachkovsky.it_anews.retrofit.NewsService
+import com.krachkovsky.it_anews.util.Constants.GENERAL_ERROR_MESSAGE
+import com.krachkovsky.it_anews.util.Constants.QUERY_PAGE_SIZE
+import com.krachkovsky.it_anews.util.Constants.RECYCLER_ITEM_SPACE
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class NewsListFragment : Fragment() {
+    private var _binding: FragmentListBinding? = null
+    private val binding
+        get() = requireNotNull(_binding) {
+            "View was destroyed"
+        }
+    private val adapter = NewsAdapter { article ->
+        findNavController().navigate(
+            NewsListFragmentDirections.actionNewsListFragmentToNewsArticleFragment(article.url)
+        )
+    }
+
+    private var currentPage = 1
+
+    var isLoading = false
+    var isLastPage = false
+    var isScrolling = false
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentListBinding.inflate(inflater, container, false)
+            .also { binding ->
+                _binding = binding
+            }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        loadNews()
+
+        with(binding) {
+//            toolbarList.menu
+//                .findItem(R.id.action_search)
+//                .let { it.actionView as SearchView }
+//                .setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+//                    override fun onQueryTextSubmit(query: String): Boolean {
+//                        return true
+//                    }
+//
+//                    override fun onQueryTextChange(newText: String): Boolean {
+//                        searchNews(newText)
+//                        Log.d("AAA search newText", newText)
+//                        return false
+//                    }
+//                })
+
+            val layoutManager = LinearLayoutManager(
+                view.context, LinearLayoutManager.VERTICAL, false
+            )
```

Comment:
можно сделать проще, у тебя используется конструктор с дефолтными параметрами (если посмотреть на конструктор)
```suggestion
            val layoutManager = LinearLayoutManager(view.context)
```